### PR TITLE
Fix stickies deflection

### LINKED ISSUE #23: Reflected stickies can kill teammates.
### Describe the bug

Sticky bombs that have been reflected by an enemy Pyro can hurt teammates when detonated.

### To Reproduce

1. Place stickies
2. Get enemy pyro to reflect them

### Expected behavior

The sticky bomb should not change team or color when an enemy pyro reflects it. It should only have a force applied.

### Media/Files

https://user-images.githubusercontent.com/91832803/185144353-5fb04ec7-3ea1-4847-ae70-d7460fcf7f2b.mp4

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bodolaz146**
# Changes
- Stickies don't change team when reflected
- Stickies can be moved by reflection even after they touched a surface
- Stickybomb launcher applies random angular velocity to the stickies

~~Owner and Launcher properties are still changed after reflect, because in base tf2 there is special killfeed icon for the case when demo detonates himself with reflected sticky. Because of that, reflected stickies still can kill teammates, as SDKGame ignores TeamNumber of projectiles.
Should I remove overriding Owner and Launcher for stickies or should some changes be made in SDKGame?~~
Now it fixes #23

https://user-images.githubusercontent.com/84805593/205511993-c5fdcb9c-96d2-414c-b44c-bc058c3e4c56.mp4

### CODE REVIEW

**Comment by Retr0-01** [comment]
Hey, @bodolaz146 

Just wanted to check in and see if you have made any progress with this PR.

**Comment by bodolaz146** [comment]
Sorry, I didn't have a lot of free time. I'll try to look into this in the next couple of days.

**Comment by bodolaz146** [comment]
I've added another check in TFGameRules.CanEntityTakeDamage, now it should fix #23

**Review by Retr0-01 — APPROVED** [review]

**Review by qualitykk — APPROVED** [review]

File: code/Player/Weapons/Projectiles/StickyBomb.cs [hunk 1]
```diff
@@ -30,7 +43,30 @@ protected override void OnPhysicsCollision( CollisionEventData eventData )
...
+			// The sticky bomb has touched a surface at least once, let's apply velocity manually
+			MoveType = ProjectileMoveType.Physics;
+
+			var vecDir = WorldSpaceBounds.Center - who.WorldSpaceBounds.Center;
+			vecDir = vecDir.Normal;
+			PhysicsBody.Velocity = vecDir * DeflectionForce;
+		}
+
+		TimeUntilRestick = Time.Now + tf_grenade_force_sleeptime;
+		TimeUntilDeflectReset = Time.Now + tf_pipebomb_deflect_reset_time;
```
Comment (rob5300) [inline code/Player/Weapons/Projectiles/StickyBomb.cs:67]: Name seems inaccurate, as they are actually the time WHEN x happens.

**Comment by Retr0-01** [comment]
Hey @bodolaz146, this PR will be put on hold for a bit until we can sort out some issues we have with the damage info system.

**Comment by Retr0-01** [comment]
> Hey @bodolaz146, this PR will be put on hold for a bit until we can sort out some issues we have with the damage info system.

This has been fixed. Please resolve your conflicts and your PR will be merged.

**Pull request merged**